### ISSUE TITLE
Atom crashes on install packages

### ISSUE THREAD

**Issue by aquavitae** [atom/atom#9055]
System details:
* Atom 1.0.19
* Windows 7

Whenever I go to "Install" to add packages, atom crashes after a couple of seconds with a windows error "The editor has crashed":
![crash](https://cloud.githubusercontent.com/assets/5753990/10302524/558a937c-6c0d-11e5-910b-ccd56f754516.png)

Yesterday I got a new machine (very similar specs) and installed atom but it still happens.  It also happens in safe mode and nothing is logged to the console. I'm not sure how to debug it further.

**Comment by subjectix** [comment]
Try running "As Administrator" it might be that atom doesn't have privilege to modify it's packages directory.

**Comment by aquavitae** [comment]
Nope, still crashes as administrator.

**Comment by subjectix** [comment]
What is the filesystem of the partition where you $HOMEPATH% is? It can be because of the long file names like: `.../node_modules/foo/node_modules/bar/node_modules/baz/...`. Not all filesystems can handle that, if it's not NTFS (for windows) you are doomed :sweat_smile:.

**Comment by aquavitae** [comment]
It looks like atom is installed to `~\AppData\local\atom`, which I'm pretty sure is NTFS.  I also tried manually deleting everything atom-related I could find to makes sure there weren't any left over node_modules, but it still crashes.  Is there a way I can force it to install to another location (e.g. `C:\atom`) with a shorter path name?

**Comment by subjectix** [comment]
with NTFS you shouldn't have that problem. what the version of npm that you are using? maybe update node and npm and reinstall atom. because the newer versions of npm try to flatten the hierarchy. These are just shots in the dark, the best I can come up with :3 I hope you can get it fixed asap.

P.S. once I couldn't get it working either. maybe try truncating the compiler-cache folder? that fixed my issue once.

**Comment by aquavitae** [comment]
Node and npm are already the latest (4.1.1 and 3.3.5 respectively), and I already cleared the entire `~\.atom` folder including compiler-cache.  One other thing I've just noticed though: I'm running behind a rather monstrous proxy with self-signed certificates and the only way I can get apm (or npm, gem, git, etc) to work is by using `strict_ssl=false` (or the equivalent for the other tools).  Before I add that line, downloading the list of available packages fails (obviously), but atom doesn't crash.  After I add the line, atom loads the list of packages, then crashes.  I'm not too sure what this says, but it might be of some use.
